source/slang/syntax.h

```diff
@@ -291,10 +291,7 @@ namespace Slang
     struct QualType
     {
         RefPtr<Type>	type;
-        bool					IsLeftValue;
-
-        template <typename T>
-        T* As();
```

Comment:
Out of curiosity, why drop this convenience method? Is it because having it use the `.as(...)` member syntax but return a raw pointer would go against the current model where `as<T>(exp)` always returns a raw pointer and `exp.as<T>()` always returns a smart pointer?

Reply:
Yes. That the idea was that use of an 'as' method returns a wrapped thing. On QualType this is a bit odd - as it seems that any code that used as didn't want or need that behavior. Well and QualType isn't a template - so it can't do. So using free function is consistent with other uses and makes it clear at the call site, that it is returning a raw pointer.